### PR TITLE
More indepth steps for selfhosting backend

### PR DESCRIPTION
**Pull request by Pokeylooted**
Adding more in depth steps to selfhost, so we can see more community supported instances and later make a hub for them all? Future plans lmao. For now helps selfhosting, I debated whether to add a guide for reverse proxy and Cloudflare Argo tunneling, you guys can lmk! https://rentry.co/Movie-Web-Backend/ has the reverse proxy part in it.

### CODE REVIEW

**Comment by Pokeylooted** [comment]
@binaryoverload Looks better now?

File: content/4.backend/1.deploy.md [hunk 1]
```diff
@@ -12,26 +12,84 @@ For configuration, check out the [configuration reference](2.configuration.md).
...
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/template/TS4mw5)
+
+1.  **Create or Log in to Railway:** Visit https://railway.app and either create an account or log in.
+2.  **Deploy with One Click:** Click the "Deploy on Railway" button above.
+3.  **Configure Environment Variables:**  Fill in the required environment variables or modify the defaults.
+4.  **Deploy:**  Click the "Deploy" button. 
+5.  **Access Your Deployment:**  Once deployment is complete, retrieve the URL from your Railway Deployments page.
+
+**Congratulations!** You've successfully deployed the backend. Proceed to [set up the client](../1.self-hosting/2.use-backend.md).
```
Comment (binaryoverload) [inline content/4.backend/1.deploy.md:92]: Duplicated Railway section

File: content/4.backend/1.deploy.md [hunk 2]
```diff
@@ -12,26 +12,84 @@ For configuration, check out the [configuration reference](2.configuration.md).
...
+          - movie-web-network
+
+      movie-web:
+        image: ghcr.io/movie-web/backend:latest
+        environment:
+          MWB_POSTGRES__CONNECTION: postgresql://movie_web_user:YourPasswordHere@postgres:5432/movie_web_backend
+          MWB_CRYPTO__SESSION_SECRET: 32CharacterLongStringHere 
+          MWB_META__NAME: unofficial-movie-web 
+          MWB_POSTGRES__MIGRATE_ON_BOOT: "true" 
+        ports:
```
Comment (binaryoverload) [inline content/4.backend/1.deploy.md:49]: I'd recommend adding this, it fixes some migration issues on certain platforms

```suggestion
          MWB_POSTGRES__MIGRATE_ON_BOOT: "true"
          MIKRO_ORM_MIGRATIONS_DISABLE_FOREIGN_KEYS: "true" 
        ports:
```

**Review by binaryoverload — CHANGES_REQUESTED** [review]

File: content/4.backend/1.deploy.md (hunk 1 above)
Comment (Pokeylooted) [reply content/4.backend/1.deploy.md]: Fixed mb

**Comment by Pokeylooted** [comment]
@binaryoverload What do you think about adding a section for cloudflare argo tunnel? Too much to add onto docs?

**Review by Pokeylooted — COMMENTED** [review]
Should've fixed it now

**Comment by binaryoverload** [comment]
> @binaryoverload What do you think about adding a section for cloudflare argo tunnel? Too much to add onto docs?

@Pokeylooted I think just change the `cloudflared tunnel` text to have a link to the docs on it

File: content/4.backend/1.deploy.md [hunk 3]
```diff
@@ -12,26 +12,71 @@ For configuration, check out the [configuration reference](2.configuration.md).
...
+    ```
+
+**Accessing Your Backend**
+
+Your backend should be accessible on `(YourPrivateIP):80`. To share it outside your local network, you'll need to configure port forwarding or cloudflared tunnel. 
+
+**Optional: Using a Reverse Proxy** 
+For SSL and domain configuration, consider setting up a reverse proxy like Nginx.
+
+* If you do use a reverse proxy, you may need to add `MWB_SERVER__CORS: "example.com movie.example.com"`.
```
Comment (binaryoverload) [inline content/4.backend/1.deploy.md:78]:
```suggestion
* If you use a reverse proxy behind a domain, add `MWB_SERVER__CORS: "https://movie.example.com"`.
  - MWB_SERVER_CORS must contain a **space separated** list of origins (Protocol + Hostname) for the client to be able to access the backend.
```

**Review by binaryoverload — CHANGES_REQUESTED** [review]

**Review by binaryoverload — DISMISSED** [review]
Looks good, thanks for your effort!

**Review by binaryoverload — APPROVED** [review]

**Pull request merged**